### PR TITLE
[FIXED JENKINS-32372] Inject the payload into the build

### PR DESCRIPTION
**Pull request by fbelzunc**
This pull request injects the payload received by BitBucket into the build through `$BITBUCKET_PAYLOAD`, which will allow users to process the payload on the build through the environment variable.

@reviewbybees

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/BitBucketTrigger.java [hunk 1]
```diff
@@ -40,7 +41,7 @@ public BitBucketTrigger() {
     /**
      * Called when a POST is made.
      */
-    public void onPost(String triggeredByUser) {
+    public void onPost(String triggeredByUser, final String payload) {
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/plugins/BitBucketTrigger.java]: Binary compatibility break

File: src/main/java/com/cloudbees/jenkins/plugins/BitbucketJobProbe.java [hunk 2]
```diff
@@ -27,7 +27,7 @@
 
 public class BitbucketJobProbe {
 
-    public void triggerMatchingJobs(String user, String url, String scm) {
+    public void triggerMatchingJobs(String user, String url, String scm, String payload) {
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/plugins/BitbucketJobProbe.java]: Binary compatibility

File: src/main/java/com/cloudbees/jenkins/plugins/BitBucketPayload.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        this.payload = payload;
+    }
+
+    @Nonnull
+    public String getPayload() {
+        return payload;
+    }
+
+    @Override
+    public void buildEnvVars(AbstractBuild<?, ?> abstractBuild, EnvVars envVars) {
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/plugins/BitBucketPayload.java]: Unfortunately such approach won't work in Workflow. Is it required?

**Comment by oleg-nenashev** [comment]
:bug: for the binary compatibility issue.
I'm also aware about Workflow compatibility

File: src/main/java/com/cloudbees/jenkins/plugins/BitBucketPayload.java (hunk 3 above)
Comment (amuniz) [inline src/main/java/com/cloudbees/jenkins/plugins/BitBucketPayload.java]: In workflow you can use `currentBuild.rawBuild.getAction(BitBucketPayload).payload` to retrieve the value.

**Comment by fbelzunc** [comment]
@oleg-nenashev @amuniz Can I have your bee now?

**Comment by amuniz** [comment]
:bee:

**Comment by fbelzunc** [comment]
@reviewbybees done

**Pull request merged**